app/services/api/record_representation_order.rb

```diff
@@ -42,10 +47,23 @@ def request_body
         applicationReference: application_reference,
         statusDate: status_date,
         effectiveStartDate: effective_start_date,
+        effectiveEndDate: effective_end_date,
         defenceOrganisation: defence_organisation
-      }
+      }.compact
     end
 
-    attr_reader :url, :status_code, :application_reference, :status_date, :effective_start_date, :defence_organisation, :headers, :connection
+    def update_database(response)
+      offence = ProsecutionCaseDefendantOffence.find_by(offence_id: offence_id)
```

Comment:
In an ideal world, `offence` would always be present. But the `#find_by` can return `nil`. Do we want to handle this?

Reply:
Good point, I've made a [follow up ticket](https://dsdmoj.atlassian.net/browse/CACP-313) for this as this calls for a bit of refactoring